plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/LibraryFactory.java

```diff
@@ -99,29 +105,64 @@ static Library create(IConfigurationElement configurationElement) throws Library
   }
 
   private static List<LibraryFile> getLibraryFiles(IConfigurationElement[] children)
-      throws InvalidRegistryObjectException, URISyntaxException {
+      throws InvalidRegistryObjectException, URISyntaxException, LibraryFactoryException {
     List<LibraryFile> libraryFiles = new ArrayList<>();
     for (IConfigurationElement libraryFileElement : children) {
       if (ELEMENT_NAME_LIBRARY_FILE.equals(libraryFileElement.getName())) {
         MavenCoordinates mavenCoordinates = getMavenCoordinates(
             libraryFileElement.getChildren(ELEMENT_NAME_MAVEN_COORDINATES));
-        LibraryFile libraryFile = new LibraryFile(mavenCoordinates);
-        libraryFile.setFilters(getFilters(libraryFileElement.getChildren()));
-        // todo do we really want these next two to be required?
-        libraryFile.setSourceUri(
-            getUri(libraryFileElement.getAttribute(ATTRIBUTE_NAME_SOURCE_URI)));
-        libraryFile.setJavadocUri(
-            getUri(libraryFileElement.getAttribute(ATTRIBUTE_NAME_JAVADOC_URI)));
-        String exportString = libraryFileElement.getAttribute(ATTRIBUTE_NAME_EXPORT);
-        if (exportString != null) {
```

Comment:
Worth considering if we should have an `exportDependencies` flag too?

Reply:
In general, what should our default be? To export dependencies or not to export them?

Reply:
Hmm, Maven projects don't export their _Maven dependencies_:

![screen shot 2017-09-01 at 12 41 47 pm](https://user-images.githubusercontent.com/202851/29979344-16a72258-8f13-11e7-8328-d96da1e1a33f.PNG)

And `WEB-INF/lib` libraries are not either for WTP _Dynamic Web Projects_:

![screen shot 2017-09-01 at 12 40 37 pm](https://user-images.githubusercontent.com/202851/29979336-0eebe1e8-8f13-11e7-96e2-63d5ca8c770b.PNG)

So I'd say no exporting by default?

Reply:
Makes sense, though it is a change to the current behavior. That's not going to change which jars we deploy to the server, right?